src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -38,13 +38,17 @@ public static class ActivityExtensions
         /// </summary>
         /// <param name="activity">Activity instance.</param>
         /// <param name="status">Activity execution status.</param>
+        [Obsolete("Use https://learn.microsoft.com//dotnet/api/system.diagnostics.activity.setstatus?view=net-7.0")]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void SetStatus(this Activity activity, Status status)
         {
             Debug.Assert(activity != null, "Activity should not be null");
 
             activity.SetTag(SpanAttributeConstants.StatusCodeKey, StatusHelper.GetTagValueForStatusCode(status.StatusCode));
             activity.SetTag(SpanAttributeConstants.StatusDescriptionKey, status.Description);
+
+            // For processors\exporters that do not look up tags
```

Comment:
why are we adding new code to the method being marked obsolete?

Reply:
Should we
1) mark method obsolete and not set the status
OR
2) set the status and mark it obsolete some time later.

Reply:
neither is needed to fix the issue described in the PR